src/main/java/gregicadditions/machines/TileEntityProcessingArray.java

```diff
@@ -356,31 +363,79 @@ protected void multiplyInputsAndOutputs(List<CountableIngredient> newRecipeInput
+			String recipeMapName = findRecipeMapName(unlocalizedName);
 
-				if(unlocalizedName.contains("gregtech.machine") || unlocalizedName.contains("gtadditions.machine")) {
-					this.numberOfMachines = Math.min(16, wholeItemStack.getCount());
-					String trimmedName;
-					String voltage = unlocalizedName.substring(unlocalizedName.lastIndexOf(".") + 1);
-					trimmedName = unlocalizedName.substring(0, unlocalizedName.lastIndexOf("."));
 
-					//Checks if the tile entity is actually a machine
-					if(!GAEnums.voltageMap.containsKey(voltage))
-						continue;
+			//Check the machine against the Config blacklist
+			if(!findMachineInBlacklist(recipeMapName)) {
 
-					this.machineName = trimmedName.substring(trimmedName.lastIndexOf(".") + 1);
-					this.machineTierVoltage = GAEnums.voltageMap.get(voltage);
-					this.machineItemStack = wholeItemStack;
-					break;
+				RecipeMap<?> rmap = RecipeMap.getByName(recipeMapName);
+
+				RecipeBuilder<?> rbuilder = rmap.recipeBuilder();
+
+				//Find the RecipeMap of the MTE and ensure that the Processing Array only works on SimpleRecipeBuilders
+				//For some reason GTCE has specialized recipe maps for some machines, when it does not need them
+				if (rbuilder instanceof SimpleRecipeBuilder ||
+						rbuilder instanceof IntCircuitRecipeBuilder ||
+						rbuilder instanceof ArcFurnaceRecipeBuilder ||
+						rbuilder instanceof CutterRecipeBuilder ||
+						rbuilder instanceof UniversalDistillationRecipeBuilder) {
+
+					return rmap;
 				}
 			}
-			return machineName;
+			return null;
+		}
+
+		public String findRecipeMapName(String unlocalizedName) {
+
+			String trimmedName = unlocalizedName.substring(0, unlocalizedName.lastIndexOf("."));
+			trimmedName = trimmedName.substring(trimmedName.lastIndexOf(".") + 1);
+
+			//Catch some cases where the machine's name is not the same as its recipe map's name
+			switch (trimmedName) {
+				case "cutter":
+					trimmedName = "cutting_saw";
+					break;
+				case "electric_furnace":
+					trimmedName = "furnace";
+					break;
+				case "ore_washer":
+					trimmedName = "orewasher";
+					break;
+			}
+
+			return trimmedName;
+		}
+
+		public boolean findMachineInBlacklist(String unlocalizedName) {
+
+			String[] blacklist = GAConfig.processingArray.machineBlackList;
+
+			return Arrays.asList(blacklist).contains(unlocalizedName);
+		}
+
+		public Tuple<ItemStack, RecipeMap<?>> findMachineStack() {
+
+			RecipeMapMultiblockController controller = (RecipeMapMultiblockController) this.metaTileEntity;
+
+			//The Processing Array is limited to 1 Machine Interface per multiblock, and only has 1 slot
+			ItemStack machine = controller.getAbilities(GACapabilities.PA_MACHINE_CONTAINER).get(0).getStackInSlot(0);
+
+			RecipeMap<?> rmap = findRecipeMapAndCheckValid(machine);
+			if(rmap == null) {
+				return null;
+			}
+
+			return new Tuple<ItemStack, RecipeMap<?>>(machine, rmap);
```

Comment:
use the diamond operator

Reply:
The whole bottom of this method could just be turned into:
```java
return (rmap == null) ? null : new Tuple<>(machine, rmap);
```

I'm not usually one to promote ternaries but this one is trivial to understand.